### ISSUE TITLE
OTA DFU Failures

### ISSUE THREAD

**Issue by AIT-tommy** [wasp-os/wasp-os#335]
The PineTime watch is stuck on the bootloader. I can see the white Pine logo, the blue Bluetooth logo, and connect on android with NRF connect. I can push .zip files from the DFU button, however, after about a few seconds (at about 26%), the kB/s drops to almost zero, hits 100%, and nothing on the watch happens. I have tried following the tutorial video for OTA flash from pinetime to waspos, and waspos to pinetime. I'm stuck on this. Anything helps, thanks.

**Comment by AIT-tommy** [comment]
This worked from InfiniTime to waspOs but not for waspOs to InfiniTime. Any Thoughts? Have been attempting SWD with ESP32 as well after breaking open the back and accessing hardware pins.

**Comment by fgaz** [comment]
you could try with a different device. you can also upload it from a computer with the dfu script in the tools directory

**Comment by fgaz** [comment]
maybe people in the irc/matrix channel can help https://github.com/daniel-thompson/wasp-os#community

**Comment by PeterCxy** [comment]
There seems to be a regression in nRF Connect v4.25.x that made the flashing process stall with wasp-os's bootloader. Downgrading to 4.24.x fixes the issue (original source: https://www.reddit.com/r/pinetime/comments/tu5z6b/stuck_in_bootloaderdfu_mode/). I ran into this exact issue on my new PineTime and I was able to get out of it by downgrading nRF Connect.

I've already added a note in the Pine64 wiki's page about switching to wasp-os, maybe this should be in the wasp-os documentation too? (or maybe we should figure out why it got stuck and actually fix it...)

**Comment by WilliamBehrens** [comment]
@PeterCxy is there a way to downgrade nrf connect from the playstore?

**Comment by PeterCxy** [comment]
No, but nRF Connect has a GitHub repository where you can find apks of older versions available for download

On June 17, 2022 12:45:17 PM EDT, William Behrens ***@***.***> wrote:
***@***.*** is there a way to downgrade nrf connect from the playstore?

**Comment by AIT-tommy** [comment]
@WilliamBehrens https://github.com/NordicSemiconductor/Android-nRF-Connect/releases

**Comment by esspe2** [comment]
> Can confirm that I was unable to access DFU mode on 4.25 until I rolled back to 4.24.3

I too had the same problem with nRF-Connect: 
I was stuck with the 4.26, so I used tools/ota-dfu/dfu.py on linux which is rather slow : 6 minutes for 300kB. 
After reading this thread, I gave a try to 4.24.3 and it works like a charm, all was sent in 1:30 minute.

Looking at the graph on my phone, v4.26.0 tries to upload at full speed > 100kBps and fails, 
while 4.24.3 starts around 14.5kBps then stabilizes around 3kBps and succeeds then the watch reboots as expected.
